engine/common/synchronization/engine.go

```diff
@@ -240,23 +212,26 @@ func (e *Engine) process(originID flow.Identifier, event interface{}) error {
 }
 
 // responseProcessingLoop is a separate goroutine that performs processing of queued responses
-func (e *Engine) responseProcessingLoop() {
+func (e *Engine) responseProcessingLoop(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
+
 	notifier := e.responseMessageHandler.GetNotifier()
+	done := ctx.Done()
 	for {
 		select {
-		case <-e.unit.Quit():
+		case <-done:
 			return
 		case <-notifier:
-			e.processAvailableResponses()
+			e.processAvailableResponses(ctx)
 		}
 	}
 }
 
 // processAvailableResponses is processor of pending events which drives events from networking layer to business logic.
-func (e *Engine) processAvailableResponses() {
+func (e *Engine) processAvailableResponses(ctx context.Context) {
```

Comment:
How about passing a channel instead? As you mentioned in another thread it actually has a performance benefit

Reply:
I have a slight preference for passing the context, because passing a context and using it this way is a very common and well-understood pattern in Go.